src/System/extract_system_rep.rs

```diff
@@ -168,3 +195,48 @@ pub fn get_system_recipe(
         _ => panic!("Got unexpected query side: {:?}", side),
     }
 }
+
+fn validate_reachability_input(
+    machine: &SystemRecipe,
+    start: &QueryExpression,
+    end: &QueryExpression,
+) {
+    let mut components: usize = 0;
+    count_component(machine, &mut components);
+
+    for (state, str) in [(start, "start"), (end, "end")] {
+        if component_to_location_count_equal(&components, state) {
+            panic!(
+                "The number of automata does not match the number of locations in the {}",
+                str
+            );
+        }
+    }
+}
+
+fn count_component(system: &SystemRecipe, count: &mut usize) {
+    match system {
+        SystemRecipe::Composition(left, right) => {
+            count_component(left, count);
+            count_component(right, count);
+        }
+        SystemRecipe::Conjunction(left, right) => {
+            count_component(left, count);
+            count_component(right, count);
+        }
+        SystemRecipe::Quotient(left, right, _) => {
+            count_component(left, count);
+            count_component(right, count);
+        }
+        SystemRecipe::Component(_) => {
+            *count += 1;
+        }
+    }
+}
+
+fn component_to_location_count_equal(components: &usize, state: &QueryExpression) -> bool {
```

Comment:
Reason for `&usize` and not just `usize`?

Reply:
Fixed